src/main/java/com/microsoft/sqlserver/jdbc/PersistentTokenCacheAccessAspect.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import com.microsoft.aad.msal4j.ITokenCacheAccessAspect;
+import com.microsoft.aad.msal4j.ITokenCacheAccessContext;
+
+
+/**
+ * Access aspect for accessing the token cache.
+ * 
+ * MSAL token cache does not persist beyond lifetime of the application. This class implements the
+ * ITokenCacheAccessAspect interface to persist the token cache between application instances so subsequent
+ * authentications can use silent authentication if the user account is in the token cache.
+ * 
+ * @see <a href="https://aka.ms/msal4j-token-cache">https://aka.ms/msal4j-token-cache</a>
+ */
+class PersistentTokenCacheAccessAspect implements ITokenCacheAccessAspect {
+    private static PersistentTokenCacheAccessAspect instance = null;
+
+    private PersistentTokenCacheAccessAspect() {};
+
+    static PersistentTokenCacheAccessAspect getInstance() {
+        if (null == instance) {
+            instance = new PersistentTokenCacheAccessAspect();
+        }
+        return instance;
+    }
+
+    /**
+     * Token cache in JSON format
+     */
+    private static String cache = null;
+
+    @Override
+    public synchronized void beforeCacheAccess(ITokenCacheAccessContext iTokenCacheAccessContext) {
+        if (null != cache && null != iTokenCacheAccessContext && null != iTokenCacheAccessContext.tokenCache()) {
+            iTokenCacheAccessContext.tokenCache().deserialize(cache);
+        }
+    }
+
+    @Override
+    public synchronized void afterCacheAccess(ITokenCacheAccessContext iTokenCacheAccessContext) {
+        if (null != iTokenCacheAccessContext && null != iTokenCacheAccessContext.tokenCache()
+                && iTokenCacheAccessContext.hasCacheChanged()) {
```

Comment:
I would reorder this to 
```java
if (null != iTokenCacheAccessContext && iTokenCacheAccessContext.hasCacheChanged()
        && null != iTokenCacheAccessContext.tokenCache())
```
since we want to know if the cache has changed first, and then if it has, then we need to do a null check on the token cache so we can call `.serialize()`.